libindy/src/services/ledger/mod.rs

```diff
@@ -542,21 +542,70 @@ impl LedgerService {
         Ok(res)
     }
 
-    pub fn parse_response<T>(response: &str) -> Result<Reply<T>, LedgerError> where T: DeserializeOwned + ReplyType {
+    pub fn get_response_metadata(&self, response: &str) -> Result<String, LedgerError> {
+        info!("get_response_metadata >>> response: {:?}", response);
+
+        let message: Message<serde_json::Value> = serde_json::from_str(response)
+            .map_err(|err|
+                LedgerError::InvalidTransaction(format!("Cannot deserialize transaction Response: {:?}", err)))?;
+
+        let response_object: Reply<serde_json::Value> = LedgerService::handle_response_message_type(message)?;
+        let response_result = response_object.result();
+
+        let response_metadata = match response_result["ver"].as_str() {
```

Comment:
Is this logic same for read and write requests?

Reply:
yes